scripts/build_gh_pages

```diff
@@ -4,22 +4,23 @@ readonly SCRIPTS_DIR="$(dirname "$(readlink -f "$0")")"
 # shellcheck source=scripts/common
 source "$SCRIPTS_DIR/common"
 
+# Create a temporary directory to stage the output of the generation process.
+readonly TARGET_DIR=$(mktemp --directory --tmpdir=/tmp 'project-oak-gh-pages-XXXXXXXXXX')
+
+# Clone the gh-pages branch to the target directory, limiting to one commit.
+git clone git@github.com:project-oak/oak.git --branch=gh-pages --depth=1 "${TARGET_DIR}"
+
+# Remove everything from the target directory. This script is supposed to automatically recreate
+# everything within that directory.
+rm -rf ${TARGET_DIR}/*
+
 readonly SRC_SUBDIRS=(sdk/rust oak/server/rust)
 readonly DOC_SUBDIRS=(sdk server)
 
 # Update the gh-pages branch. Note that `cargo doc` is **not deterministic** so
 # this should only be done when there is a real change.
 readonly BRANCH=${1:-master}
-readonly PREV_COMMIT=$(git log --oneline -n 1 gh-pages | sed 's/.*branch at \([0-9a-f]*\)/\1/')
```

Comment:
We should probably also set `git config core.abbrev` to a standard value in this script for consistency; 12 is what's been used for all previous `gh-pages` commit messages.

Reply:
Done now using `--short=12` flag (which I just found out about).